core/commonMain/src/implementations/immutableList/AbstractPersistentList.kt

```diff
@@ -7,21 +7,12 @@ package kotlinx.collections.immutable.implementations.immutableList
 
 import kotlinx.collections.immutable.ImmutableList
 import kotlinx.collections.immutable.PersistentList
-import kotlinx.collections.immutable.mutate
 
 abstract class AbstractPersistentList<E> : PersistentList<E>, AbstractList<E>() {
     override fun subList(fromIndex: Int, toIndex: Int): ImmutableList<E> {
         return super<PersistentList>.subList(fromIndex, toIndex)
     }
 
-    override fun addAll(elements: Collection<E>): PersistentList<E> {
-        return mutate { it.addAll(elements) }
-    }
-
-    override fun addAll(index: Int, c: Collection<E>): PersistentList<E> {
-        return mutate { it.addAll(index, c) }
-    }
```

Comment:
Why are `addAll`'s moved down by hierarchy?

Reply:
To avoid creating builder in `SmallPersistentVector` when `this.size + c.size <= MAX_BUFFER_SIZE`

Reply:
`SmallPersistentVector` overrides them anyway, right? But for other implementations this seems to be a sensible default.

Reply:
Right.
`addAll` methods are returned.